File: contracts/wasm/gascalibration/executiontime/go/executiontime/executiontime.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package executiontime

import "github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib"

func funcF(ctx wasmlib.ScFuncContext, f *FContext) {
	n := f.Params.N().Value()
	x := uint32(0)
	y := uint32(0)

	for i := uint32(0); i < n; i++ {
		x++
		y = 3 * (x % 10)
	}

	f.Results.N().SetValue(y)
}

File: contracts/wasm/gascalibration/executiontime/src/executiontime.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_f(_ctx: &ScFuncContext, f: &FContext) {
    let n = f.params.n().value();
    let mut x: u32 = 0;
    let mut y: u32 = 0;

    for _ in 0..n {
        x += 1;
        y = 3 * (x % 10);
    }

    f.results.n().set_value(y)
}
